fgpyo/sam/__init__.py

```diff
@@ -863,6 +863,28 @@ def sum_of_base_qualities(rec: AlignedSegment, min_quality_score: int = 15) -> i
     return score
 
 
+def _set_common_mate_fields(dest: AlignedSegment, source: AlignedSegment) -> None:
+    """Set common mate info an alignment to its mate's primary alignment.
+
+    Args:
+        dest: The alignment to set the mate info upon.
+        source: The primary alignment to use as a mate reference.
+    """
+    if source.is_read1 is dest.is_read1 or source.is_secondary or source.is_supplementary:
```

Comment:
I think we could/should remove `source.is_secondary` from this `if`.

Reply:
Yes, good find!